.cicd/build.sh

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+set -eo pipefail
+. ./.cicd/helpers/general.sh
+
+mkdir -p $BUILD_DIR
+
+if [[ $(uname) == 'Darwin' ]]; then
+
+    # You can't use chained commands in execute
+    cd $BUILD_DIR
+    [[ $TRAVIS == true ]] && ccache -s
+    cmake ..
+    make -j$JOBS
+
+else # Linux
+
+    ARGS=${ARGS:-"--rm --init -v $(pwd):$MOUNTED_DIR"}
+
+    . $HELPERS_DIR/docker-hash.sh
+
+    PRE_COMMANDS="cd $MOUNTED_DIR/build"
+    # PRE_COMMANDS: Executed pre-cmake
+    # CMAKE_EXTRAS: Executed within and right before the cmake path (cmake CMAKE_EXTRAS ..)
+    if [[ $IMAGE_TAG == 'ubuntu-18.04' ]]; then
+        PRE_COMMANDS="$PRE_COMMANDS && export PATH=/usr/lib/ccache:\\\$PATH"
+        CMAKE_EXTRAS="$CMAKE_EXTRAS -DCMAKE_CXX_COMPILER='clang++' -DCMAKE_C_COMPILER='clang' -DCMAKE_BUILD_TYPE='Release'"
+    elif [[ $IMAGE_TAG == 'ubuntu-16.04' ]]; then
+        PRE_COMMANDS="$PRE_COMMANDS && export PATH=/usr/lib/ccache:\\\$PATH"
+        CMAKE_EXTRAS="$CMAKE_EXTRAS -DCMAKE_TOOLCHAIN_FILE=$MOUNTED_DIR/.cicd/helpers/clang.make -DCMAKE_CXX_COMPILER_LAUNCHER=ccache"
+    elif [[ $IMAGE_TAG == 'centos-7.6' ]]; then
+        PRE_COMMANDS="$PRE_COMMANDS && source /opt/rh/devtoolset-8/enable && source /opt/rh/rh-python36/enable && export PATH=/usr/lib64/ccache:\\\$PATH"
+    elif [[ $IMAGE_TAG == 'amazon_linux-2' ]]; then
+        PRE_COMMANDS="$PRE_COMMANDS && export PATH=/usr/lib64/ccache:\\\$PATH"
+        CMAKE_EXTRAS="$CMAKE_EXTRAS -DCMAKE_CXX_COMPILER='clang++' -DCMAKE_C_COMPILER='clang'"
+    fi
+
+    BUILD_COMMANDS="cmake $CMAKE_EXTRAS -DBUILD_MONGO_DB_PLUGIN=true .. && make -j$JOBS"
+
+    # Docker Commands
+    if [[ $BUILDKITE == true ]]; then
+        # Generate Base Images
+        $CICD_DIR/generate-base-images.sh
+        [[ $ENABLE_INSTALL == true ]] && COMMANDS="cp -r $MOUNTED_DIR ~/eosio && cd ~/eosio/build &&"
```

Comment:
Is there a way we can specify a directory, rather than relying on HOME?  I would rather fail in the pipelines if the directory doesn't exist than push images that are inconsistent with others.

Reply:
Great idea! Adding